core/compiler/src/main/scala/com/fortysevendeg/exercises/compiler/codeGen.scala

```diff
@@ -0,0 +1,142 @@
+package com.fortysevendeg.exercises
+package compiler
+
+import scala.language.implicitConversions
+import scala.reflect.api.Universe
+
+case class CodeGen[U <: Universe](override val u: U = scala.reflect.runtime.universe) extends TreeHelpers[U] {
+  import u._
+
+  def emitExercise(
+    name: Option[String]
```

Comment:
This formatting is weird. Is this Scalariform? We should tweak the rules for something more sane if so.